tests/garage/tf/policies/test_deterministic_mlp_policy_with_model_transit.py

```diff
@@ -0,0 +1,90 @@
+"""
+Unit test for DeterministicMLPPolicyWithModel.
+
+This test consists of four different DeterministicMLPPolicy: P1, P2, P3
+and P4. P1 and P2 are from DeterministicMLPPolicy, which does not use
+garage.tf.models.DeterministicMLPModel while P3 and P4 do use.
+
+This test ensures the outputs from all the policies are the same,
+for the transition from using DeterministicMLPPolicy to
+DeterministicMLPPolicyWithModel.
+
+It covers get_action, get_actions, get_action_sym.
+"""
+from unittest import mock
+
+import numpy as np
+import tensorflow as tf
+
+from garage.tf.envs import TfEnv
+from garage.tf.policies import DeterministicMLPPolicy
+from garage.tf.policies import DeterministicMLPPolicyWithModel
+from tests.fixtures import TfGraphTestCase
+from tests.fixtures.envs.dummy import DummyBoxEnv
+
+
+class TestDeterministicMLPPolicyWithModelTransit(TfGraphTestCase):
+    @mock.patch('tensorflow.random.normal')
+    def setUp(self, mock_rand):
+        mock_rand.return_value = 0.5
+        super().setUp()
+        self.box_env = TfEnv(DummyBoxEnv())
```

Comment:
make sure this doesn't always produce zero-observations (or your tests aren't testing anything)